added startup check and additional option + docs

wsdookadr opened the pull request:
The following Perl program was written. The `max_memory_size` will intentionally not be met in order to trigger the shut down of the process.

```perl
#!/usr/bin/perl
use lib qw(lib);
use Mojolicious::Lite;
use Mojolicious::Plugin::SizeLimit;
plugin 'Mojolicious::Plugin::SizeLimit', 
    { 
        max_process_size => (1024**2) * 0.01,
        startup_check    => 1,
    };
get '/' => sub {
  my $c = shift;
};
app->start;
```

Fixes #2 

Please consider merging #5 before this, in order for the tests to pass.

Commit:
added startup check and additional option + docs

Fixes #2

Commit:
fixed typo

augensalat reviewed (changes requested):
Hi, thanks again.
Unfortunately this is a total "doing-it-wrong".
1. You must never `exit` the Mojolicious process! You find the correct way for process termination in the original code in line 71.
2. You are doubling code. All check-code is already there. The challenge is to run the `_limits_are_exceeded` function at start, and change this function that if might be given a `$count` argument of `0` or `undef`, which then must be handled in a sensible way.

Commit:
refactored _make_limits_are_exceeded and _make_count_requests ; added tests for startup_check parameter

wsdookadr commented:
> Unfortunately this is a total "doing-it-wrong".

Yes, that is true, it was completely wrong.

> You must never exit the Mojolicious process!

Yup, I'm using `stop_gracefully` now

> You are doubling code [..] The challenge is to run the _limits_are_exceeded function at start, and change this function

Yes, I've [completely rewritten the functions  `_make_limits_are_exceeded` and `_make_count_requests`](https://github.com/augensalat/mojolicious-plugin-sizelimit/pull/6/commits/beb09f00281d5ddb6432bb91e2bc06cb50252f6e), and added docs for them. I've also added a test for `startup_check`.

If #5 is merged first, and then this one #6 , then all tests will pass. Do note that none of the tests were modified/altered in any way, except for minor API updates in #5 , which means all functionality/behaviour stays the same.

Commit:
improved test message

wsdookadr commented:
> I will have a look at the problem later or in the next days.

Hi @augensalat 

Is anything else needed here? Can I help you with anything for this pull request to get merged?

wsdookadr commented:
Closing due to maintainer being unresponsive
